src/core/init/load_on_media_source.ts

```diff
@@ -157,36 +159,33 @@ export default function createMediaSourceLoader(
             log.debug("Init: resume-stream order received.");
             cancelEndOfStream$.next(null);
             return EMPTY;
-          case "discontinuity-encountered":
-            const { bufferType, gap } = evt.value;
-            if (SegmentBuffersStore.isNative(bufferType)) {
-              handleDiscontinuity(gap[1], mediaElement);
-            }
+          case "stream-status":
+            const { period, bufferType, imminentDiscontinuity, position } = evt.value;
+            discontinuities$.next({ period,
```

Comment:
So here, the discontinuities subject is 'nexted' even if there is no imminent discontinuity ?

Reply:
I'm not sure to understand the way it was done. The discontinuities event can contain no discontinuity and tells here information about the current buffer position, type and content? I don't find it very clear.

Reply:
> So here, the discontinuities subject is 'nexted' even if there is no imminent discontinuity ?
Yes.

The problem is that we want to catch the possible case where a discontinuity "disappear".

That's why we also emit when there's no discontinuity: the `StallAvoider` might want to remove a previously advertised discontinuity for that type and period.

Reply:
Ok, but why not keeping a subject name closer to streamStatus$ or something?

Reply:
I see what you mean. I though that `streamStatus$` might tie the StallAvoider to much to the current `RepresentationStream` implementation, where `discontinuities$` is a more neutral term for what it actually is.

But it's true that it is more difficult to "get" the fact that those are to be updated each time a new one is received for the same type and Period.

Reply:
Renamed to `discontinuityUpdate$`